dlt/destinations/impl/databricks/sql_client.py

```diff
@@ -0,0 +1,166 @@
+
+from dlt.common import pendulum
+from dlt.common import logger
+from dlt.common.destination import DestinationCapabilitiesContext
+from dlt.destinations.exceptions import (
+    DatabaseTerminalException,
+    DatabaseTransientException,
+    DatabaseUndefinedRelation,
+)
+from dlt.destinations.sql_client import (
+    DBApiCursorImpl,
+    SqlClientBase,
+    raise_database_error,
+    raise_open_connection_error,
+)
+from dlt.destinations.typing import DBApi, DBApiCursor, DBTransaction
+from dlt.destinations.impl.databricks.configuration import DatabricksCredentials
+from dlt.destinations.impl.databricks import capabilities
+from dlt.common.time import to_py_date, to_py_datetime
+
+
+class DatabricksSqlClient(SqlClientBase[DatabricksSqlConnection], DBTransaction):
+    dbapi: ClassVar[DBApi] = databricks_lib
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(self, dataset_name: str, credentials: DatabricksCredentials) -> None:
+        super().__init__(credentials.catalog, dataset_name)
+        self._conn: DatabricksSqlConnection = None
+        self.credentials = credentials
+
+    def open_connection(self) -> DatabricksSqlConnection:
+        conn_params = self.credentials.to_connector_params()
+        self._conn = databricks_lib.connect(**conn_params, schema=self.dataset_name)
+        return self._conn
+
+    @raise_open_connection_error
+    def close_connection(self) -> None:
+        if self._conn:
+            self._conn.close()
+            self._conn = None
+
+    @contextmanager
+    def begin_transaction(self) -> Iterator[DBTransaction]:
+        # Databricks does not support transactions
+        yield self
+
+    @raise_database_error
+    def commit_transaction(self) -> None:
+        # Databricks does not support transactions
+        pass
+
+    @raise_database_error
+    def rollback_transaction(self) -> None:
+        # Databricks does not support transactions
+        pass
+
+    @property
+    def native_connection(self) -> "DatabricksSqlConnection":
+        return self._conn
+
+    def drop_dataset(self) -> None:
+        self.execute_sql("DROP SCHEMA IF EXISTS %s CASCADE;" % self.fully_qualified_dataset_name())
+
+    def drop_tables(self, *tables: str) -> None:
+        # Tables are drop with `IF EXISTS`, but databricks raises when the schema doesn't exist.
+        # Multi statement exec is safe and the error can be ignored since all tables are in the same schema.
+        with suppress(DatabaseUndefinedRelation):
+            super().drop_tables(*tables)
+
+    def execute_sql(
+        self, sql: AnyStr, *args: Any, **kwargs: Any
+    ) -> Optional[Sequence[Sequence[Any]]]:
+        with self.execute_query(sql, *args, **kwargs) as curr:
+            if curr.description is None:
+                return None
+            else:
+                f = curr.fetchall()
+                return f
+
+    def execute_many(
```

Comment:
ok, now I see why you had to override further up. maybe this also is more a case for a capabilty for sql destinations, I am not 100% sure though.

Reply:
Actually you have it with supports_many... why not use that in the super class?